native/runtime/ppx_deriving_router_primitives.ml

```diff
@@ -60,3 +60,19 @@ let option_of_url_query f k xs =
         match f k [ k, v ] with
         | Ok v -> Ok (Some v)
         | Error err -> Error err))
+
+let list_to_url_query f k xs =
+  List.fold_left (fun acc x -> List.rev_append (f k x) acc) [] xs
+  |> List.rev
```

Comment:
Yeah, I think order is important, even if not to rely on but just for debugging sanity. Also it seems like we can use `List.concat_map` which is supported on all OCaml version this lib supports (>=4.14):
```suggestion
let list_to_url_query f k xs = List.concat_map (f k) xs
```